atakama/packager.py

```diff
@@ -0,0 +1,246 @@
+
+    A package consists of wheels, signatures and a cert.
+
+    Wheels are just unpacked into the plugins/ folder.
+    """
+
+    def __init__(
+        self, src=None, pkg=None, key=None, crt=None, self_signed=False, openssl=None
+    ):
+        self.pkg = pkg
+        self.src = src
+        self.key = key
+        self.crt = crt
+        self.self_signed = self_signed
+
+        self.setup_path = src and os.path.abspath(os.path.join(self.src, "setup.py"))
+        self.openssl_path = openssl or which("openssl")
+        self.openssl_path = self.openssl_path and os.path.abspath(self.openssl_path)
+        self.made_setup = False
+
+    @classmethod
+    def from_args(cls, argv):
+        """Given sys args, return a packager."""
+        args = cls.parse_args(argv)
+        return Packager(
+            src=args.src,
+            pkg=args.pkg,
+            key=args.key,
+            crt=args.crt,
+            self_signed=args.self_signed,
+            openssl=args.openssl,
+        )
+
+    @staticmethod
+    def parse_args(argv):
+        """Command line argument parser."""
+        parser = argparse.ArgumentParser(
+            description="Atakama plugin packaging helper",
+            epilog="""
+
+        An atakama plugin package consists of a python installable package, an openssl signature file,
+        and a certificate.
+
+        These three files are located in the same zip.
+
+        It is installed by installing the package into the "plugins" folder.
+
+        The python package can be:
+            - a simple zip of sources
+            - a binary wheel (which is treated, for now, as a zip anyway)
+
+        This tool simply shells out to openssl as needed to produce the signature.
+        """,
+        )
+
+        parser.add_argument("--src", help="Package source root folder.")
+        parser.add_argument("--pkg", help="Package file path (wheel, for example)")
+        parser.add_argument("--key", help="Openssl private key file", required=True)
+        parser.add_argument("--crt", help="Openssl certificate file", required=True)
+        parser.add_argument(
+            "--openssl", help="Location of openssl binary", default=which("openssl")
+        )
+        parser.add_argument(
+            "--self-signed", help="Allow a self-signed cert", action="store_true"
+        )
+        args = parser.parse_args(argv)
+        if not args.src and not args.pkg:
+            raise ValueError("Nothing to do: must specify --src or --pkg")
+        return args
+
+    def run_setup(self) -> None:
+        """Given a src directory, run the setup.py command within it."""
+        wd = os.getcwd()
+        try:
+            os.chdir(os.path.dirname(self.src))
+            print("running: ", os.getcwd(), self.setup_path, file=sys.stderr)
+            dist = run_setup(self.setup_path, script_args=["bdist_wheel"])
+            for typ, _, path in getattr(dist, "dist_files"):
```

Comment:
`dist.dist_files`?

Reply:
was 4 no-member lint.  it's a documented property.   no idea.   can replace with a #pylint thing